src/frontend/src/common/components/library/data_subview/components/CreateNSTreeModal/components/SampleFiltering/components/CollectionDateFilter/index.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from "react";
+import { getDateRangeLabel } from "src/common/utils/dateUtils";
+import { MENU_OPTION_ALL_TIME } from "src/components/DateFilterMenu/constants";
+import { StyledDateFilterMenu, StyledInputDropdown } from "./style";
+
+interface Props {
+  fieldKeyEnd: string;
+  fieldKeyStart: string;
+  menuOptions: DateMenuOption[];
+  updateDateFilter: UpdateDateFilterType;
+}
+
+const CollectionDateFilter = ({
+  updateDateFilter,
+  ...props
+}: Props): JSX.Element => {
+  // `startDate` and `endDate` represent the active filter dates. Update on filter change.
+  const [startDate, setStartDate] = useState<FormattedDateType>();
+  const [endDate, setEndDate] = useState<FormattedDateType>();
+  const [anchorEl, setAnchorEl] = useState<HTMLElement | undefined>();
```

Comment:
i've seen you use anchorEl multiple times, but i'm still not 100% on what it's doing, do you mind explaining?

Reply:
InputDropdown is an element that is always inline in the page. The menu sometimes appears in the page, but sometimes doesn't. When it _does_ appear, it does not occur inline with the other elements, it floats on top. Because it's a floating element and can't position itself based on the other elements around it, it needs to know the exact location on the page where it is supposed to float. The `anchorEl` (InputDropdown) is the element that the menu uses to anchor itself. That is, it's the element that the menu uses to position itself (by relatively offsetting its own position based on the position of the anchor).

Reply:
i wonder why i've never had to use it. what do you mean by `The menu sometimes appears in the page, but sometimes doesn't` you mean visually when a user opens the modal?

Reply:
> you mean visually when a user opens the modal?

Yep!

If you use and SDS menu or dropdown, you don't need to use it because it's an implementation detail that is handled by the library. In this case, we're implementing the two pieces separately so we control the relationship between the two.

Reply:
@phoenixAja If you open the developer console and try to follow the elements, you'll see that the thing opening the menu and the menu itself are on totally different chains of DOM nodes. The menu isn't exactly owned in the traditional `<div><span> I am a child of span and a grandchild of div!! </span></div>` kind of way. The menu just lives outside of that chain of nodes entirely, waiting to be called into existence, then gets anchored to the page at the appropriate spot by that anchorEl.